src/NsisoLauncherCore/Auth/YggdrasilAuthenticator.cs

```diff
@@ -123,144 +88,53 @@ public class YggdrasilTokenAuthenticator : IAuthenticator
-                    validate.Arguments = AuthArgs;
-                }
-                var resultTask = validate.PerformRequestAsync();
-                var result = resultTask.Result;
+                var validate = new Validate(AccessToken);
+                if (ProxyAuthServerAddress != null) validate.Address = new Uri(ProxyAuthServerAddress + "/validate");
+                if (AuthArgs != null && AuthArgs.Count != 0) validate.Arguments = AuthArgs;
+                var result = await validate.PerformRequestAsync();
                 if (result.IsSuccess)
                 {
                     return new AuthenticateResult(AuthState.SUCCESS)
-                    {
-                        AccessToken = this.AccessToken,
-                        SelectedProfileUUID = this.SelectedProfileUUID,
-                        UserData = this.UserData
-                    };
+                        {AccessToken = AccessToken, UserData = UserData, SelectedProfileUUID = SelectedProfileUUID};
                 }
-                else
-                {
-                    AuthState state;
-                    Refresh refresh = new Refresh(AccessToken);
-                    if (ProxyAuthServerAddress != null)
-                    {
-                        validate.Address = new Uri(ProxyAuthServerAddress + "/refresh");
-                    }
-                    if (AuthArgs != null && AuthArgs.Count != 0)
-                    {
-                        refresh.Arguments = AuthArgs;
-                    }
-                    var refreshResultTask = refresh.PerformRequestAsync();
-                    var refreshResult = refreshResultTask.Result;
-                    if (refreshResult.IsSuccess)
-                    {
-                        this.AccessToken = refreshResult.AccessToken;
-                        state = AuthState.SUCCESS;
-                    }
-                    else
-                    {
-                        state = AuthState.REQ_LOGIN;
-                        if (refreshResult.Code == System.Net.HttpStatusCode.NotFound)
-                        { state = AuthState.ERR_NOTFOUND; }
-                    }
-                    return new AuthenticateResult(state)
-                    {
-                        AccessToken = AccessToken = this.AccessToken,
-                        SelectedProfileUUID = this.SelectedProfileUUID,
-                        UserData = this.UserData,
-                        Error = refreshResult.Error
-                    };
-                }
-            }
-            catch (Exception ex)
-            {
-                return new AuthenticateResult(AuthState.ERR_INSIDE)
-                {
-                    Error = new Net.MojangApi.Error() { ErrorMessage = ex.Message, Exception = ex },
-                    AccessToken = AccessToken = this.AccessToken,
-                    UserData = this.UserData,
-                    SelectedProfileUUID = this.SelectedProfileUUID
-                };
-            }
```

Comment:
This code is used to refresh the Access token, should not be deleted.

Reply:
The code resolver auto merge some same state to avoid nesting.